### ISSUE TITLE
Argument missing and ordering mismatch at `torch.distributed.rpc.init_model_parallel` doc

### ISSUE THREAD

**Issue by Kytabyte** [pytorch/pytorch#29905]
## 📚 Documentation

Per `torch.distributed.rpc.init_model_parallel` function on RPC document page in master branch [here](https://pytorch.org/docs/master/rpc.html), I think there are some points to revise or polish:

1. the argument `worker_name_to_id` is missing in doc.
2. The order of arguments in doc does not match the function signature, which would cause additional ambiguity or inconvenience for readers.
3. I think the repr for argument `backend` is too long to read, which may contain a lot of unneeded information for readers. It is arguable but it might be better to give the class a concise repr.

cc @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @xush6528

**Comment by Kytabyte** [comment]
I can try to fix this if it's needed and it's not urgent, as I have no experience in contributing PyTorch but I'm willing to take the first stab

**Comment by mrshenli** [comment]
Hey @Kytabyte, thanks a lot for working on this!! The `rpc` API is currently experimental and we are still working on documents. Regarding `worker_name_to_id`, we are removing this arg and it won't appear in the first experimental release, and `init_model_parallel` API will be renamed to `init_rpc` (#29762).

**Comment by mrshenli** [comment]
Per 3. the repr for `backend` is indeed an issue. I see the following API signature:

```
torch.distributed.rpc.init_model_parallel(self_name, backend=<BackendType.PROCESS_GROUP: BackendValue(init_backend_handler=<function _process_group_init_backend_handler>)>, init_method=None, self_rank=-1, worker_name_to_id=None, num_send_recv_threads=4, rpc_timeout=datetime.timedelta(0, 60))
```

@rohan-varma @xush6528 Shall we stay consistent with [`init_process_group`](https://pytorch.org/docs/stable/distributed.html#torch.distributed.init_process_group) and just show `backend` in the signature?

**Comment by rohan-varma** [comment]
@mrshenli I agree, though one difference is `init_rpc` provides a default backend type and `init_process_group` does not provide a default backend arg, hence the discrepancy in the docs. We should probably try to change the repr that is showing up to something simpler like just `BackendType.PROCESS_GROUP` like @Kytabyte suggested.

**Comment by mrshenli** [comment]
#30093 addressed the missing arg issue and #30243 fixes the backend type repr issue.
